lib/core/src/Cardano/Pool/DB/Sqlite.hs

```diff
@@ -208,7 +209,8 @@ newDBLayer trace fp = do
             let poolPledge_ = getQuantity poolPledge
             let poolMetadataUrl = fst <$> poolMetadata
             let poolMetadataHash = snd <$> poolMetadata
-            insert_ $ PoolRegistration
+            deleteWhere [PoolOwnerPoolId ==. poolId]
```

Comment:
The delete is redundant with the use of `repsert`, isn't ?

Reply:
No, this is removing the pool _owners_ (before we later `insertMany_ `)

The `repsert` is used with the pool _registration certificate_

Reply:
However — I think my solution here is flawed in the context of rollbacks :/

We need to preserve the old certificate, in case the new certificate gets rolled back.

Maybe we can get away with something simple though. Like just keeping both and reading the latest one, when we need to read it.

Reply:
Indeed.. rollbacks. What about simply changing the PrimaryKey to:

```
    Primary poolRegistrationPoolId poolRegistrationSlot
```

In the rare case we have two certificates at the same slot, then only the latest one will prevail (we still need the repsert here) as it should, and it'll preserve old certificates on past slots if there are any. 

We'll have to be careful when _reading_ the certificates from the db now, and make sure they're ordered by descending slot. As for the owners :/ ... it's more complicated.

Reply:
> As for the owners :/ ... it's more complicated

Could we not store the owners as some kind of comma-separated list as part of the `pool_registration` table?

Reply:
…or perhaps that's not possible.

Reply:
:/ ... What about using MongoDB then? 

Not really, this is not quite the problem, and it won't solve the need for a migration. Ideally, we should also attach a slot to the pool owner, or a certificate number, so that when we fetch a certificate, we can fetch _just_ the owner corresponding to that certificate.

Reply:
Are you sure it is possible to use both slot and poolId as primary key, while using them as a foreign key at the same time?

I added (to the `PoolRegistration` table):
```
    Foreign PoolRegistration fk_registration_pool_id poolOwnerPoolId
    Foreign PoolRegistration fk_registration_slot poolOwnerPoolSlot
```

But get:
```

found 1 fkeys and 2 pkeys: fdef=ForeignDef {foreignRefTableHaskell = HaskellName {unHaskellName = "PoolRegistration"}, foreignRefTableDBName = DBName {unDBName = "pool_registration"}, foreignConstraintNameHaskell = HaskellName {unHaskellName = "fk_registration_pool_id"}, foreignConstraintNameDBName = DBName {unDBName = "pool_ownerfk_registration_pool_id"}, foreignFields = [], foreignAttrs = [], foreignNullable = False} pdef=CompositeDef {compositeFields = [FieldDef {fieldHaskell = HaskellName {unHaskellName = "poolRegistrationPoolId"}, fieldDB = DBName {unDBName = "pool_id"}, fieldType = FTTypeCon (Just "W") "PoolId", fieldSqlType = SqlOther "SqlType unset for poolRegistrationPoolId", fieldAttrs = ["sql=pool_id"], fieldStrict = True, fieldReference = NoReference, fieldComments = Nothing},FieldDef {fieldHaskell = HaskellName {unHaskellName = "poolRegistrationSlot"}, fieldDB = DBName {unDBName = "slot"}, fieldType = FTTypeCon (Just "W") "SlotId", fieldSqlType = SqlOther "SqlType unset for poolRegistrationSlot", fieldAttrs = ["sql=slot"], fieldStrict = True, fieldReference = NoReference, fieldComments = Nothing}], compositeAttrs = []}
```

Reply:
Yes, you need the foreign key to match the primary key of the foreign table. See: https://github.com/input-output-hk/cardano-wallet/pull/1798/commits/81fe0c38b2db9a928c91af52b47729c70cc72438

Reply:
I think this should do.